File: OOP/vizsga_minta/Konyv.cpp
#include "Konyv.h"
#include <string>
#include <iostream>

using namespace std;

Konyv::Konyv(string szerzo_neve, string konyv_cime, int kiadas_eve)
	: szerzo(szerzo_neve), cim(konyv_cime), kiadas(kiadas_eve) {}

Konyv::~Konyv(){}

string Konyv::getSzerzo() const {
	return szerzo;
}

string Konyv::getCim() const {
	return cim;
}

int Konyv::getKiadas() const
{
	return kiadas;
}

void Konyv::print() {
	cout << szerzo << " - " << cim << "(" << kiadas << ")" << endl;
}

string Konyv::format() const {
	return szerzo + " - " + cim + "(" + to_string(kiadas) + ")";
}

ostream & operator<<(ostream & cout, const Konyv& konyv)
{
	cout << konyv.format();
	return cout;
}

File: PHP/5_gyak/Konyv.php
<?php

class Konyv{
    private $cim;
    private $szerzo;
    private $ISBN;
    private $kulcsszavak;
    
    public function __construct($cim, $szerzo, $ISBN, $kulcsszavak) {
        $this->cim = $cim;
        $this->szerzo = $szerzo;
        try{
            if(strlen($ISBN) != 13){
                throw new Exception ("Az ISBN 13 hosszu legyen!");
            }
            $this->ISBN = $ISBN;
        } catch (Exception $ex) {
            echo $ex->getMessage();
        }
        $this->ISBN = $ISBN;
        $this->kulcsszavak = $kulcsszavak;
    }
    
    public function getCim(){
        return $this->cim;
    }
    
    public function getSzerzo(){
        return $this->szerzo;
    }
    
    public function getISBN(){
        return $this->ISBN;
    }
    
    public function getKulcsszavak(){
        return $this->kulcsszavak;
    }
    
    public function setCim($cim){
        $this->cim=$cim;
    }
    
    public function setSzerzo($szerzo){
        $this->szerzo=$szerzo;
    }
    
    public function setISBN($ISBN){
        try{
            if(strlen($ISBN) != 13){
                throw new Exception ("Az ISBN 13 hosszu legyen!");
            }
            $this->ISBN = $ISBN;
        } catch (Exception $ex) {
            echo $ex->getMessage();
        }
    }
    
    public function setKulcsszavak($kulcsszavak){
        $this->kulcsszavak=$kulcsszavak;
    }
    
    public function __toString() {
        $s= "{$this->szerzo}:{$this->cim}({$this->ISBN}): ";
        foreach ($this->kulcsszavak as $ksz) {
            $s .= $ksz." ";
        }
        return $s;
    }
    
    public function kulcsszavakSzama($megadott){
        $i=0;
        foreach ($megadott as $ksz) {
            if(in_array($ksz, $this->kulcsszavak)){
                $i++;
            }
        }
        return $i;
    }
}

